conjure-java-core/src/integrationInput/java/com/palantir/product/UnionTypeExample.java

```diff
@@ -124,6 +125,140 @@ public String toString() {
+
+        private Function<Integer, T> thisFieldIsAnIntegerVisitor;
+
+        private Function<Integer, T> alsoAnIntegerVisitor;
+
+        private Function<Integer, T> ifVisitor;
+
+        private Function<Integer, T> newVisitor;
+
+        private Function<Integer, T> interfaceVisitor;
+
+        public SetStageVisitorBuilder<T> stringExample(
+                Function<StringExample, T> stringExampleVisitor) {
+            this.stringExampleVisitor = stringExampleVisitor;
+            return this;
+        }
+
+        public ThisFieldIsAnIntegerStageVisitorBuilder<T> set(Function<Set<String>, T> setVisitor) {
+            this.setVisitor = setVisitor;
+            return this;
+        }
+
+        public AlsoAnIntegerStageVisitorBuilder<T> thisFieldIsAnInteger(
+                Function<Integer, T> thisFieldIsAnIntegerVisitor) {
+            this.thisFieldIsAnIntegerVisitor = thisFieldIsAnIntegerVisitor;
+            return this;
+        }
+
+        public IfStageVisitorBuilder<T> alsoAnInteger(Function<Integer, T> alsoAnIntegerVisitor) {
+            this.alsoAnIntegerVisitor = alsoAnIntegerVisitor;
+            return this;
+        }
+
+        public NewStageVisitorBuilder<T> if_(Function<Integer, T> ifVisitor) {
+            this.ifVisitor = ifVisitor;
+            return this;
+        }
+
+        public InterfaceStageVisitorBuilder<T> new_(Function<Integer, T> newVisitor) {
+            this.newVisitor = newVisitor;
+            return this;
+        }
+
+        public UnknownStageVisitorBuilder<T> interface_(Function<Integer, T> interfaceVisitor) {
+            this.interfaceVisitor = interfaceVisitor;
+            return this;
+        }
+
+        public Visitor<T> unknown(Function<String, T> unknownVisitor) {
+            return new Visitor<T>() {
+                public T visitStringExample(StringExample value) {
+                    return stringExampleVisitor.apply(value);
+                }
+
+                public T visitSet(Set<String> value) {
+                    return setVisitor.apply(value);
+                }
+
+                public T visitThisFieldIsAnInteger(int value) {
+                    return thisFieldIsAnIntegerVisitor.apply(value);
+                }
+
+                public T visitAlsoAnInteger(int value) {
+                    return alsoAnIntegerVisitor.apply(value);
+                }
+
+                public T visitIf(int value) {
+                    return ifVisitor.apply(value);
+                }
+
+                public T visitNew(int value) {
+                    return newVisitor.apply(value);
+                }
+
+                public T visitInterface(int value) {
+                    return interfaceVisitor.apply(value);
+                }
+
+                public T visitUnknown(String value) {
+                    return unknownVisitor.apply(value);
+                }
+            };
+        }
+    }
+
+    public interface StringExampleStageVisitorBuilder<T> {
```

Comment:
this seems like a cumbersome pattern. You probably want to only implement a `from` method. What is the use-case you had in mind?

Reply:
Could you clarify? How would that `from` method look?

Reply:
Do you mean something like the `of(...)` method for immutables? Like

    Visitor<String> from(Function<String, String> catVisitor,
                         Function<String, String> dogVisitor,
                         Function<String, String> unknownVisitor) {
        ...
    }

?